app/src/main/java/com/expidev/gcmapp/MainActivity.java

```diff
@@ -394,7 +406,26 @@ void showEditTraining(final long trainingId)
 
     public void goToMeasurements(MenuItem menuItem)
     {
-        startActivity(new Intent(getApplicationContext(), MeasurementsActivity.class));
+        if(mAssignment != null && mAssignment.getRole() == Assignment.Role.BLOCKED)
```

Comment:
for code readability it might be beneficial to provide some boolean accessors on the Assignment class for the various roles. `if(mAssignment != null && mAssignment.isBlocked())`